webdriver/tests/bidi/__init__.py

```diff
@@ -0,0 +1,22 @@
+# Compares 2 objects recursively ignoring values of specific attributes.
+def recursive_compare(expected, actual, ignore_attributes=None):
+    if ignore_attributes is None:
+        ignore_attributes = []
+    assert type(expected) == type(actual)
+    if type(expected) is list:
+        assert len(expected) == len(actual)
+        for index, val in enumerate(expected):
+            recursive_compare(expected[index], actual[index], ignore_attributes)
+        return
+
+    if type(expected) is dict:
+        assert expected.keys() == actual.keys(), \
+            f"Key sets should be the same: " \
+            f"\nNot present: {set(expected.keys()) - set(actual.keys())}" \
+            f"\nUnexpected: {set(actual.keys()) - set(expected.keys())}"
+        for index, val in enumerate(expected):
+            if val not in ignore_attributes:
```

Comment:
Instead of just ignoring some fields, I think we should allow the expeced value to be a callable that is used to validate the return value. Then we could predefine helpers like `any_string` which would just be

```
def any_string(expected):
    return isinstance(expected, str)
```

That would allow us to write custom validators in the future e.g. for strings that must be a UUID or positive integers, or so on.

Reply:
Good idea. Done.